### PR TITLE
Vector class with arithmetics

### PR DESCRIPTION
**Pull request by zuntrax**
I'll need this in my C++ perlin implementation.

TODO:
* [x] add cross_product
* [x] add test
* [x] final cleanup and squash

### CODE REVIEW

**Comment by zuntrax** [comment]
@fr3akout suggested that i make a test case and add a cross_product as special case for N==3

File: libopenage/util/vector.h [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+		for (int i = 0; i < N; i++) {
+			(*this)[i] += other[i];
+		}
+		return *this;
+	}
+
+	/**
+	 * Vector addition
+	 */
+	Vector<N> operator+(const Vector<N> &other) const {
```
Comment (TheJJ) [inline libopenage/util/vector.h]: returns `Vector<N> &` as well

File: libopenage/util/vector.h [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+ * Vector class with arithmetic.
+ */
+template<size_t N>
+class Vector : public std::array<float, N> {
+public:
+
+	/**
+	 * Vector addition with assignment
+	 */
+	Vector<N>& operator+=(const Vector<N> &other) {
```
Comment (TheJJ) [inline libopenage/util/vector.h]: "Correct" style would be `Vector<N> &operator +=(const...`

File: libopenage/util/vector.h [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+		return *this;
+	}
+
+	/**
+	 * Cross-product of two 3-dimensional vectors
+	 */
+	template<typename T=Vector<N>>
+	typename std::enable_if<N==3,T>::type
+	/*Vector<N>*/ cross_product(const Vector<N> &other) const {
+		std::cout << "it werks" << std::endl;
```
Comment (mic-e) [inline libopenage/util/vector.h]: does it?

File: libopenage/util/vector.h [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+ */
+template<size_t N>
+class Vector : public std::array<float, N> {
+public:
+	static_assert(N > 0, "0-dimensional vector not allowed");
+
+	/**
+	 * Default constructor
+	 */
+	Vector() {};
```
Comment (fr3akout) [inline libopenage/util/vector.h]: Vector() = default;

**Comment by fr3akout** [comment]
Have you thought about copy- and move-semantics for your vector class?

**Comment by zesterer** [comment]
Just use GLM? It's perfect for this kind of stuff, 2D or 3D.

**Comment by zuntrax** [comment]
@zesterer We will discuss it.

**Comment by TheJJ** [comment]
We need vectors anyway, and if simple vec operations are not enough we can still switch to glm.

**Pull request merged**